app/assets/javascripts/discourse/app/components/search-menu.js

```diff
@@ -50,13 +44,41 @@ export default class SearchMenu extends Component {
   @tracked suggestionKeyword = false;
   @tracked suggestionResults = [];
   @tracked invalidTerm = false;
+  @tracked menuPanelOpen = false;
+
   _debouncer = null;
   _activeSearch = null;
 
-  get animationClass() {
-    return this.site.mobileView || this.site.narrowDesktopView
-      ? "slide-in"
-      : "drop-down";
+  @bind
+  setupEventListeners() {
+    document.addEventListener("mousedown", this.onDocumentPress, true);
+    document.addEventListener("touchend", this.onDocumentPress, {
+      capture: true,
+      passive: true,
+    });
+  }
+
+  willDestroy() {
+    document.removeEventListener("mousedown", this.onDocumentPress);
+    document.removeEventListener("touchend", this.onDocumentPress);
+
+    super.willDestroy(...arguments);
+  }
+
+  @bind
+  onDocumentPress(event) {
+    if (!event.target.closest(".search-menu-container.menu-panel-results")) {
+      this.menuPanelOpen = false;
+    }
+  }
+  get classNames() {
```

Comment:
style nitpick - newline between functions

```suggestion
  }

  get classNames() {
```